tests/Sitemap/SitemapFactoryTest.php

```diff
@@ -0,0 +1,110 @@
+        $cacheItemInterface->method('isHit')->willReturn(true);
+        $cacheItemInterface->method('get')->willReturn((string) new Sitemap());
+
+        $this->cacheItemPoolInterface->method('getItem')->willReturn($cacheItemInterface);
+        $this->cacheItemPoolInterface->expects($this->never())->method('save');
+
+        $this->createReflectionMethod('createContentSitemap');
+    }
+
+    public function testCreateContentSitemapWithHit()
+    {
+        $cacheItemInterface = $this->createMock(CacheItemInterface::class);
+        $cacheItemInterface->expects($this->once())->method('isHit')->willReturn(false);
+        $cacheItemInterface->expects($this->once())->method('set');
+        $cacheItemInterface->method('get')->willReturn((string) new Sitemap());
+
+        $this->cacheItemPoolInterface->method('getItem')->willReturn($cacheItemInterface);
+        $this->cacheItemPoolInterface->expects($this->once())->method('save')->with($cacheItemInterface);
+
+        $this->createReflectionMethod('createContentSitemap');
+    }
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->category = new ArticleCategory();
+        $this->category->setSlug('category');
+
+        $this->article = new Article();
+        $this->article->setSlug('article');
+        $this->article->setCategory($this->category);
+        $this->article->setUpdatedAt(new \DateTime());
+
+        $this->objectManager = $this->createMock(ObjectManager::class);
+        $this->routerInterface = $this->createMock(RouterInterface::class);
+        $this->cacheItemPoolInterface = $this->createMock(CacheItemPoolInterface::class);
+
+        $categoriesRepository = $this->createMock(ArticleCategoryRepository::class);
+        $categoriesRepository->method('findAll')->willReturn([$this->category]);
+
+        $articleRepository = $this->createMock(ArticleRepository::class);
+        $articleRepository->method('findAllPublished')->willReturn([$this->article]);
+
+        $orderArticleRepository = $this->createMock(OrderArticleRepository::class);
+        $orderArticleRepository->method('findAllPublished')->willReturn([]);
+
+        $this->objectManager->method('getRepository')->will($this->returnValueMap([
+            [ArticleCategory::class, $categoriesRepository],
+            [Article::class, $articleRepository],
+            [OrderArticle::class, $orderArticleRepository],
+        ]));
+    }
+
+    protected function tearDown()
+    {
+        $this->objectManager = null;
+        $this->routerInterface = null;
+        $this->cacheItemPoolInterface = null;
+        $this->article = null;
+        $this->category = null;
+
+        parent::tearDown();
+    }
+
+    private function createReflectionMethod(string $methodName, Sitemap $sitemap = null)
+    {
+        $reflection = new \ReflectionMethod(SitemapFactory::class, $methodName);
+        $reflection->setAccessible(true);
```

Comment:
This feels really hacky to me, unit tests should not care about private methods.

Reply:
The card of this PR is to test these privates functions. Is it better to change the visibility of these methods ?

Reply:
No don't change the visibility of the methods because the tests look weird. The reflection classes are made for tests, hacks and hacky edge cases.

Reply:
Private methods are implementation details that we should not care in a test. In test we should rely on public API.
If it's needed to change the visibility of the methods to do the right test, it just means that the design is seriously bad.

Reply:
I don't agree regarding unit tests everything can/should be tested.

Reply:
Yes but through public API.